### ISSUE TITLE
Torrents in progress (downloading) don't resume automatically on container restart

### ISSUE THREAD

**Issue by timkelty** [haugene/docker-transmission-openvpn#40]
* If a torrent is in progress...
* I stop/start my container
* if I go to the web UI now, the in progress torrent is paused, and any progress it had is lost.

Any ideas?

**Comment by timkelty** [comment]
I discovered if I "Verify Local Data", then "Resume", everything works as expected. But I believe this is supposed to happen automatically?

Is it perhaps a timing issue with when Transmission starts and when the mount points are available?

**Comment by haugene** [comment]
It's good at least that it works with "Verify Local Data". But I agree, Transmission should remember it's state from before the shutdown.

The docker engine will make mount points available before starting the openvpn process, so this is not a timing issue. 

I'm thinking this has to do with how Transmission is stopped when you do "docker stop <container>". The openvpn cofigs instructs openvpn to kill transmission when the interface goes down, this is to prevent traffic when no openvpn connection is active. When transmission is killed it might lose state that is in-memory. To fix this we should probably look into a more graceful way of shutting down Transmission, but kill the process if it doesn't stop gracefully.

**Comment by timkelty** [comment]
Obviously not a huge deal, and certainly wouldn't want to compromise the killing of transmission when the interface goes down.

Thanks for the explanation!

**Comment by haugene** [comment]
No problem. I'll keep it in mind, and maybe I'll stumble upon a better solution some day ;)
I'm closing this issue as a "won't fix" for now, and rephrased it a little to pinpoint what's the possible improvement here.

**Comment by fletom** [comment]
This seems like a huge issue to me, since it's necessary to configure automatic restarts when the VPN disconnects. So effectively a disconnect will actually not be recovered from properly, since all your torrents will be paused. This never happens when e.g. the desktop version of Transmission gets restarted.

Can we reopen this discussion and look at possible fixes?

Is it possible that there's another cache/config/tmp directory that Transmission uses for this kind of state that we can just mount as a volume to preserve between restarts?

**Comment by fletom** [comment]
It seems like this issue is also a duplicate of #23, which was supposed to have been fixed by @dougnukem's commits. Are others still seeing this issue, or is it possible that it's an issue with my own configuration?

**Comment by simonsjw** [comment]
I see the same thing.
